### PR TITLE
CNF-11871: Allow user overrides

### PR DESCRIPTION
**Pull request by nocturnalastro**

### CODE REVIEW

**Comment by lack** [comment]
I think the change looks great, but probably necessitates some doc changes too?

File: pkg/compare/testdata/UserOverride/localout.golden [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Diff Output: None
+Patched with testdata/UserOverride/localnewOverridesout.golden
+Patch Reasons: <None given>
+
+**********************************
+
+Summary
+CRs with diffs: 1/2
+No CRs are missing from the cluster
+No CRs are unmatched to reference CRs
```
Comment (lack) [inline pkg/compare/testdata/UserOverride/localout.golden:46]: Maybe the Summary should also include a line like `"%d CRs are locally patched"` or something like that?

File: pkg/compare/testdata/UserOverride/localout.golden [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
++  finalizers:
++  - kubernetes
+
+**********************************
+
+Cluster CR: v1_Namespace_openshift-storage
+Reference File: namespace.yaml
+Diff Output: None
+Patched with testdata/UserOverride/localnewOverridesout.golden
+Patch Reasons: <None given>
```
Comment (lack) [inline pkg/compare/testdata/UserOverride/localout.golden:39]: Should we maybe fail in case of "None given" to force some documentation of the patches?

**Review by lack — CHANGES_REQUESTED** [review]
Generally really good, but we need doc updates too before we merge this please!

File: pkg/compare/testdata/UserOverride/localout.golden (hunk 2 above)
Comment (nocturnalastro) [reply pkg/compare/testdata/UserOverride/localout.golden:39]: Do you mean fail to generate them or fail to report on a diff?

File: pkg/compare/testdata/UserOverride/localout.golden (hunk 1 above)
Comment (nocturnalastro) [reply pkg/compare/testdata/UserOverride/localout.golden:46]: I like that idea:)

File: pkg/compare/testdata/UserOverride/localout.golden (hunk 2 above)
Comment (nocturnalastro) [reply pkg/compare/testdata/UserOverride/localout.golden:39]: We could refuse to add the patch if no reason is given and then put an error/warning log to explain

**Comment by nocturnalastro** [comment]
@lack I added some docs in explaining something I realised I'd missed at test so I added that and fixed the bug it showed :).

File: pkg/compare/testdata/UserOverride/localout.golden (hunk 1 above)
Comment (nocturnalastro) [reply pkg/compare/testdata/UserOverride/localout.golden:46]: Done

File: pkg/compare/testdata/UserOverride/localout.golden (hunk 2 above)
Comment (lack) [reply pkg/compare/testdata/UserOverride/localout.golden:39]: I still think forcing "some string" is better than allowing nothing.  We can't stop people from lying about the reason, but we can at least encourage some kind of good behavior.

**Review by lack — APPROVED** [review]
Looks great!

**Pull request merged**